### PR TITLE
fix(Core/ScriptsMgr): correct hooks name

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Added `virtual` for `CanSendJoinMessageArenaQueue` and `CanSendExitMessageArenaQueue`
- Rename `CanExitJoinMessageArenaQueue` to `CanSendExitMessageArenaQueue`

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
@Yehonal please check

File: doc/changelog/pendings/changes_1624109504178983100.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Added
+- new hook `CanSendJoinMessageArenaQueue` and `CanSendExitMessageArenaQueue`
+
+### Changed
+- Rename `CanExitJoinMessageArenaQueue` to `CanSendExitMessageArenaQueue`
+
+### How to upgrade
+- Just rename all hooks from `CanExitJoinMessageArenaQueue` to `CanSendExitMessageArenaQueue`
```
Comment (Yehonal) [inline doc/changelog/pendings/changes_1624109504178983100.md:8]:
```suggestion
- Just rename all hooks from `CanExitJoinMessageArenaQueue` and `CanSendMessageArenaQueue`, to `CanSendExitMessageArenaQueue`
```

Comment (Yehonal) [reply doc/changelog/pendings/changes_1624109504178983100.md:8]: The original hook was called CanSendMessageArenaQueue, some modules are still using that one

**Comment by Yehonal** [comment]
Based on our [naming conventions](https://www.azerothcore.org/wiki/hooks-script#naming-conventions) the hook names should follow the structure: `On[When]<Action>`

Since we're doing a renaming here I'd suggest starting using the correct convention and documenting them properly.

Here you are an example of a proper named and documented hook for the SendExitMessageArenaQueue (the same should be done for the other one):

```cpp
/**
 * @brief This hook runs before sending the exit message during the arena queue, allowing you to run extra operations or disabling the exit message
 *
 * @param queue Contains information about the Arena queue
 * @param ginfo Contains information about the group of the queue
 * @return True if you want to continue sending the message, false if you want to disable the message
 */
[[nodiscard]] virtual bool OnBeforeSendExitMessageArenaQueue(BattlegroundQueue* /*queue*/, GroupQueueInfo* /*ginfo*/) { return true; }
};
```

NOTE: I'm not sure about what the  ginfo param contains though, feel free to change it.

**Comment by Yehonal** [comment]
@Winfidonarleyan can you complete these, please? so we can merge it

**Comment by Winfidonarleyan** [comment]
@Yehonal done, need re-review

File: doc/changelog/pendings/changes_1624109504178983100.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+### Added
+- new hook `OnBeforeSendJoinMessageArenaQueue` and `OnBeforeSendExitMessageArenaQueue`
+
+### Changed
+- Rename `CanExitJoinMessageArenaQueue` to `OnBeforeSendExitMessageArenaQueue`
+- Rename `CanSendJoinMessageArenaQueue` to `OnBeforeSendJoinMessageArenaQueue`
+
+### How to upgrade
+- Just rename all hooks from `CanExitJoinMessageArenaQueue` and `CanSendMessageArenaQueue`, to `CanSendExitMessageArenaQueue`
```
Comment (Yehonal) [inline doc/changelog/pendings/changes_1624109504178983100.md:9]:
```suggestion
- Just rename all hooks from `CanExitJoinMessageArenaQueue` and `CanSendMessageArenaQueue`, to `OnBeforeSendExitMessageArenaQueue`
```

**Review by Yehonal — CHANGES_REQUESTED** [review]

**Review by Yehonal — APPROVED** [review]

**Pull request merged**